sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveReferencesInAggregate.scala

```diff
@@ -112,9 +105,12 @@ object ResolveReferencesInAggregate extends SQLConfHelper
     assert(selectList.forall(_.resolved))
     if (conf.groupByAliases) {
       groupExprs.map { g =>
-        g.transformWithPruning(_.containsPattern(UNRESOLVED_ATTRIBUTE)) {
+        g.transformWithPruning(_.containsAnyPattern(UNRESOLVED_ATTRIBUTE,
+          LATERAL_COLUMN_ALIAS_REFERENCE)) {
           case u: UnresolvedAttribute =>
             selectList.find(ne => conf.resolver(ne.name, u.name)).getOrElse(u)
+          case u: LateralColumnAliasReference =>
+            selectList.find(ne => conf.resolver(ne.name, u.name)).getOrElse(u)
```

Comment:
nit: you could merge those two cases via `u @ (_: UnresolvedAttribute | _: LateralColumnAliasReference)`

Reply:
Nice catch, my mistake.

Reply:
I'm not sure where is the problem? It seems that scala will only recognize the public parent class of `extends`, and the public class of `with` will not be recognized
![image](https://github.com/apache/spark/assets/32387433/8a4ffb2c-0908-430a-93eb-9a8991c03b00)

Reply:
hmm, both have the common parent `NamedExpression`. You can explicitly tell to the compiler:
```scala
          case u @ (_: UnresolvedAttribute | _: LateralColumnAliasReference) =>
            selectList.find(ne => conf.resolver(ne.name, u.asInstanceOf[NamedExpression].name))
              .getOrElse(u)
```

Reply:
Got it. Thanks for remind. 😁